File: src/questions/parkinglot/java/entity/Vehicle.java
package questions.parkinglot.java.entity;

public abstract class Vehicle {

    private final String licensePlate;
    private final VehicleSize size;

    public Vehicle(String licensePlate, VehicleSize size) {
        if (licensePlate == null || licensePlate.isEmpty() || size == null) {
            throw new IllegalArgumentException("License plate cannot be null or empty");
        }
        this.licensePlate = licensePlate;
        this.size = size;
    }

    public String getLicensePlate() {
        return licensePlate;
    }

    public VehicleSize getSize() {
        return size;
    }

    @Override
    public String toString() {
        return "Vehicle{" +
                "licensePlate='" + licensePlate + '\'' +
                ", size=" + size +
                '}';
    }
}

File: src/questions/parkinglot/py/entity/Vehicle.py
from abc import ABC, abstractmethod
from questions.py.entity.VehicleSize import VehicleSize

class Vehicle(ABC):
    def __init__(self, license_plate: str, size: VehicleSize):
        if not license_plate or not license_plate.strip():
            raise ValueError("License plate cannot be empty.")
        self._license_plate = license_plate
        self._size = size

    @property
    def license_plate(self) -> str:
        """Returns the license plate of the vehicle."""
        return self._license_plate
    
    @property
    def size(self) -> VehicleSize:
        """Returns the size of the vehicle."""
        return self._size
    
    def __str__(self) -> str:
        return f"{self.__class__.__name__}[{self._license_plate}]"